### DISCUSSION TITLE
How can I apply drift_analog_weights in the forward path during training?

### DISCUSSION THREAD

**Discussion by matifali** [IBM/aihwkit#427] [Q&A]
I am trying to train a network where I want to apply the `drift_analog_weights` in the `forward` path but only want to use it for computing the `forward` pass and not update(drift) the weights for `update`.

Can it be done with the current setup?

**Comment by maljoras** [comment]
This is not possible directly, but that is the reason of the `rpu_config.modifier` settings (see [here](https://aihwkit.readthedocs.io/en/latest/api/aihwkit.simulator.configs.utils.html#aihwkit.simulator.configs.utils.WeightModifierParameter)). There a noise can be defined that is applied during forward (and backward) pass, but not on the updated weights. This noise is drawn again for each mini-batch.  Currently only a programming noise polynomial can be given, but not the exact drift (however, we plan to update this in future).
